### ISSUE TITLE
[java] CloseResource false positive when resource included in return value

### ISSUE THREAD

**Issue by OverDrone** [pmd/pmd#2470]
**Affects PMD Version:**  ???

**Rule:** [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

**Description:**

false positive when resource included in return value.

**Code Sample demonstrating the issue:**

Spring MVC controller method:

```java
public ResponseEntity<StreamingResponseBody> test() throws IOException {
  final InputStream is = Files.newInputStream(Paths.get("..."));
  return ResponseBody.ok(os -> {
    try {
      is.transferTo(os);
    } finally {
      is.close();
    }
  });
}
```

**Expected outcome:**

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]* ????
